add-on/src/lib/ipfs-proxy/index.js

```diff
@@ -0,0 +1,137 @@
+'use strict'
+/* eslint-env browser */
+
+const browser = require('webextension-polyfill')
+const { createProxyServer, closeProxyServer } = require('ipfs-postmsg-proxy')
+const AccessControl = require('./access-control')
+
+const ACL_FUNCTIONS = [
+  'block.put',
+  'config.set',
+  'config.get',
+  'config.replace',
+  'dag.put',
+  'dht.put',
+  'dht.provide',
+  'files.add',
+  'key.get',
+  'key.list',
+  'key.rename',
+  'key.rm',
+  'object.new',
+  'object.put',
+  'object.patch.addLink',
+  'object.patch.rmLink',
+  'object.patch.appendData',
+  'object.patch.setData',
+  'pin.add',
+  'pin.rm',
+  'pubsub.publish',
+  'swarm.connect',
+  'swarm.disconnect'
+]
+
+function createIpfsProxy (getIpfs, getState) {
```

Comment:
This function is doing some intricate plumbing. It's very cool. Please add some docs to pin down some of the helpful background context that is in your head so future us can pick it up.

Things like "this is when a tab is created, but also _caveat_, _caveat_" and "This end is talking from proxy to ipfs node, this end is content script to proxy.", those sort of things.